### DISCUSSION TITLE
[Export-service] Is it possible to choose the format of a date field and the timezone?

### DISCUSSION THREAD

**Discussion by benedettoraviotta** [mia-platform/community#351] [Q&A] [answered]
### Which section?

Marketplace

### Which is the version?

2.0.1

### Description

When I download an Excel file from Backoffice, for example:
![image](https://github.com/mia-platform/community/assets/59974565/b1efcf79-3d2e-477b-bb21-7ccab8a4ac0a)
date type fields are saved as a string in UTC, as show below:
![image](https://github.com/mia-platform/community/assets/59974565/bf2cbc73-3b42-4b68-82c0-65c8e8815230)

### Actual Outcome

Columns of data type in excel file are saved in format YYYY-MM-DDThh:mm:ss:mmsZ where Z is UTC timezone. 

### Expected Outcome

1. The possibility to choose the date format, for example i want that the date has format: 
YYYY-MM-DD 

2. That the time of the date respects the time zone of the user who exported the file. The date displayed by backoffice (image 1 in the Description section) is consistent with my timezone, I would like the data fields in the excel file (image 2 in the Description section) to also respect the same timezone, in this case UTC+1.

**Comment by fredmaggiowski** [comment]
If I remember correctly, a similar topic ha already been discussed and I think that a custom solution was implemented.

Regardless of this, let's discuss on this and see if there is room for some new feature in our marketplace. Let me start with a question, why the export service and not the crud service which owns the data model definition?

**Reply by benedettoraviotta** [reply]
I think the responsibility of timezone management is in a gray area.
Going back to the example above, image 1, the timezone is managed a by the backit bk-table and not by the crud service.
It is also true that the export service is not aware of the data model unlike the backit. So I think it makes more sense for the crud service to handle it. 
Might it make sense to be able to specify to the crud service, for the export endpoint, to use a certain timezone and date format?

**Reply by fredmaggiowski** [reply, marked as answer]
What comes to my mind is pretty much the same, some queryparam (only available to export API? possibly to all GET APIs?) that allows to specify the timezone (e.g. `_t=UTC+2`)

I'd suggest you to open a feature request on the [CRUD Service](https://github.com/rond-authz/rond) repository.
